### ISSUE TITLE
[BUG] Can't change language in startmenu searchbar

### ISSUE THREAD

**Issue by bdh1993** [linuxmint/cinnamon#11115]
```
 * Cinnamon version : 5.4.11
 * Distribution - Mint 21
 * 64 bit
 ```

## **Can't change language in startmenu searchbar & Alt + F2 command input bar**

![menu](https://user-images.githubusercontent.com/58254473/186585669-3929933c-ebb9-4e7f-8ba5-40435e2de8d9.gif)

![001](https://user-images.githubusercontent.com/58254473/186586847-912e04a3-1e3e-4118-b344-95a87c7b7154.png)
![003](https://user-images.githubusercontent.com/58254473/186586854-b97e5139-829e-4d4f-bdf1-25d009680de1.png)

## 1) I tested it with various input method editor(IME).
But the language hasn't changed.
- nimf
- kime
- xim
- fcitx
- ibus

## 2) I tested it in a different language.

- korean
- japanese

## 3) Tested on a different menu applet. (cinamenu)

![002](https://user-images.githubusercontent.com/58254473/186586825-8cccc97f-a34d-4410-a2df-5adba25433e9.png)

### Steps to reproduce
1) Install input method editor
2) Install language
3) test language input

### Expected behaviour
- The changed language shall be entered.

### Other information
```
 * It worked well in linuxmint 20.2.
 * cinnamon version 5.0.7
 * 64bit
 ```

**Comment by Icy-Breath** [comment]
I have the same issue.
The issue also happens in Timeshift dialog.
Using:
Linux Mint 21 
Cinnamon : 5.4.11

**Comment by vitneum** [comment]
I am also having the same issue.

I've been using it since mint version 18
This is the first time I'm experiencing this in version 21.

**Comment by davictor233** [comment]
I have the same issue.
I tested multiple distros on vbox.
Debian 11 with Cinnamon 4.8.6: working
Mint 21 with Cinnamon 5.6.5: not working
Fedora 37 and Arch with Cinnamon: not working

**Comment by AmenomoriYuuna** [comment]
> I have the same issue. I tested multiple distros on vbox. Debian 11 with Cinnamon 4.8.6: working Mint 21 with Cinnamon 5.6.5: not working Fedora 37 and Arch with Cinnamon: not working

me too

mint20.1 with cinnamon 4.8.6:working
manjaro with cinnamon 5.6.8:not warking

**Comment by fredcw** [comment]
If this started with mint version 20. it may something to do the upgrade to clutter 7?

**Comment by hyunki** [comment]
Today I upgrade from 20 to Mint 21
Have same problem.
Cinamon 5.4.12
Kernel 5.15.0.79
In addition, monitor auto brightness function also gone. It is really dark.

**Comment by thickmelon** [comment]
Cinnamon 6.0.4 
Kernel 5.15.0.91
Does not work. 
I tried iBus, like what some users said. This did not work either. 
I can't for the love of God search for any applications in my own search bar.

**Comment by lwmv** [comment]
Mint 22 x86_64
Cinnamon 6.2.7 
Kernel 6.8.0-39
fcitx-5.1.7
Exactly same issue. No language other than English can't be input in the menu search bar.

**Comment by duzhuoshanwai** [comment]
same problem here. fcitx5/ibus
I try on Debian sid,linux mint Wilma.

**Comment by Chillbok** [comment]
same problem here. using fcitx

I tried on LMDE 6(faye), x86_64
Kernel 6.1.0-12-amd64
Cinnamon 6.2.9

**Comment by thickmelon** [comment]
Either this gets fixed or we're just going to have to be stuck with this inconvenience. Harmoikr has the same issue afaik because it's a distro of Mint. AND IT'S SUPPOSED TO BE FOR KOREANS.

**Comment by zhyhang** [comment]
same issue on linux-mint 22.1

**Comment by doctorgreen4721** [comment]
The same issue
Operating system: Linux Mint 22.1 Cinnamon
Cinnamon version: 6.4.6

**Comment by DreamFerry** [comment]
The same issue
Operating system: Linux Mint 22.1 Cinnamon
Cinnamon version: 6.4.8

The issue may not be severe, but we are inevitably troubled by this inconvenience during use. I have noticed multiple similar feedback regarding this problem, which has persisted since Mint version 21 and has not been addressed for several years. I hope the project maintainers can take note of this issue.

**Comment by janos-r** [comment]
It would be nice to have a reply from some dev that understands why this issue exists.
The same issue is reported at least also in https://github.com/linuxmint/cinnamon/issues/11713 and https://github.com/linuxmint/cinnamon/issues/11610
Especially because it worked in the past before something about the whole menu changed. I would feel better about recommending Mint to Japanese users. Mint has probably the best input metod installer, but it doesn't work in the main menu.

**Comment by kayodde** [comment]
Same issue 
 * Cinnamon version : 6.4.10
 * Distribution - Arch Linux
 * 64 bit

**Comment by NULL0xFF** [comment]
> same here, LMDE7

Another LMDE 7 user here.

Seems like the patch has been merged to main branch weeks ago, so we may expect it to be released at some point in the future.
In the mean time, you will have to build by self I guess.

#12758

**Comment by evergreen-serg** [comment]
Mint 22.2 x86_64
Kernel: 6.8.0-88-generic
Cinnamon 6.4.8

I’m experiencing the same issue: the menu opens only when I release the Windows key, and it immediately closes as soon as I press the key (before releasing it). Because of this behavior, I’m unable to switch keyboard layouts while typing in the Start menu search bar.

**Comment by janos-r** [comment]
> Mint 22.2 x86_64 Kernel: 6.8.0-88-generic Cinnamon 6.4.8
> 
> I’m experiencing the same issue: the menu opens only when I release the Windows key, and it immediately closes as soon as I press the key (before releasing it). Because of this behavior, I’m unable to switch keyboard layouts while typing in the Start menu search bar.

Seams like you have the shortcut bound to a super+something combination. That's not the cause of this issue. If you would bind a different shortcut, it would still not work. And that is the issue. That this field doesn't allow it. It's not about opening/closing the menu.

**Comment by duzhuoshanwai** [comment]
Seems it fixed after my Mint zena 22.3

**Comment by evergreen-serg** [comment]
> Seams like you have the shortcut bound to a super+something combination. That's not the cause of this issue. If you would bind a different shortcut, it would still not work. And that is the issue. That this field doesn't allow it. It's not about opening/closing the menu.


> Seems it fixed after my Mint zena 22.3

It’s unclear in which direction to investigate this issue.
The menu disappears immediately when the **Win** key is pressed, not when the key is released.

I assigned **Ctrl+Tab** to open the Cinnamon menu, and this allows me to switch the keyboard layout in the search bar from **EN** to **RU** using **Win+Space** without issues. However, I would like the menu to **open on Win key press** and **close only after pressing and releasing Win again**. Otherwise, switching the keyboard layout is impossible.

I’ve also updated to **Zena**, but the issue persists.

* Linux Mint 22.3
* Cinnamon: 6.6.4
* Kernel: 6.14.0-37-generic
* X11

**Comment by zhyhang** [comment]
Upgrade to 22.3 yesterday, the problem is resolved!

<img width="440" height="254" alt="Image" src="https://github.com/user-attachments/assets/7116e262-0558-4513-b78f-0352b0533541" />

**Comment by gdemin** [comment]
It works for me before 6.6.4. Currently, in 6.6.4 I can switch layout in start menu search bar only sometimes (Ru-En, Caps Lock key).

**Comment by good-vi** [comment]
Fix that: https://github.com/linuxmint/cinnamon/pull/13699

**Comment by internet-coffee** [comment]
Has this problem been resolved? I installed Linux Mint 22.3 two days ago and confirmed that all software and drivers are up-to-date, but I still can't type Chinese in the Start Menu.
IME:Fcitx5-chewing

<img width="1766" height="513" alt="Image" src="https://github.com/user-attachments/assets/924f3c62-2aa5-4869-ae44-9a5a8321d192" />

**Comment by janos-r** [comment]
> Has this problem been resolved? IME:Fcitx5-chewing

You have to either use ibus and have the setup through the system keyboard layouts app.
Or uninstall the fcitx that installs automatically through the input methods app and install fcitx5. I recommend the second option and I use it with Japanese with mozc and everything works.
Haven't tried with chinese, but if you uninstalled all of the old fcitx and you have fcitx5 with the right layout, and switching works in a text file, it should switch in the start menu as well.

**Comment by baihongru** [comment]
> Upgrade to 22.3 yesterday, the problem is resolved!
> 
> <img alt="Image" width="440" height="254" src="https://private-user-images.githubusercontent.com/5225836/534859769-7116e262-0558-4513-b78f-0352b0533541.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzY4NzA3MTcsIm5iZiI6MTc3Njg3MDQxNywicGF0aCI6Ii81MjI1ODM2LzUzNDg1OTc2OS03MTE2ZTI2Mi0wNTU4LTQ1MTMtYjc4Zi0wMzUyYjA1MzM1NDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA0MjJUMTUwNjU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGRlY2Y1ODk3NWVhM2RkMGIxMDQ1ODJlMmU3ZjZkNjg5ZTUyMWMxNDgwODkzNzE5Mjg2ODUwZjViZDQ0ZDc5YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.mQPcOKOCHxAPbhscz_CnkGyfyH6KD-cZdWjhIgZw_wI">

I installed it recently and this problem still persists.

**Comment by Gosh42** [comment]
> You have to either use ibus and have the setup through the system keyboard layouts app.

I had the issue too and this put me on what looks like the right track. I'm like a very casual user and maybe this isn't entirely correct but I'll post how I fixed it in my case in case it helps someone

What I did was go to System Settings > Keyboard and:
1. In "XKB Settings" tab had my language switch key (Caps Lock) bound to "Switching to another layout" (already had it like that before, and it worked fine everywhere except while Start Menu was open, requiring me to close it to switch language)
2. In the "Layouts" tab, scrolled down to "Shortcuts" and bound the key as "Switch to next layout". It bound as "Next Group" instead of "Caps Lock" and now it works in Start Menu just fine  

I also removed it from "XKB Options" as a test, and it still let me bind it in Layouts, but it shows up as "Caps Lock" instead of "Next Group" and makes it so you switch the language while STILL switching the letter capitalisation, so don't remove it from XKB Options if you're using the Caps Lock as well

- OS: Linux Mint 22.3 - Cinnamon 64-bit
- Kernel: 7.0.0-28-generic
- Desktop Environment: Cinnamon 6.6.9
